AEPMessaging/Sources/Messaging+EdgeEvents.swift

```diff
@@ -0,0 +1,220 @@
+        // Create the profile experience event to send the push notification details with push token to profile
+        let profileEventData: [String: Any] = [
+            MessagingConstants.PushNotificationDetails.PUSH_NOTIFICATION_DETAILS: [
+                [MessagingConstants.PushNotificationDetails.APP_ID: appId,
+                 MessagingConstants.PushNotificationDetails.TOKEN: token,
+                 MessagingConstants.PushNotificationDetails.PLATFORM: platform,
+                 MessagingConstants.PushNotificationDetails.DENYLISTED: false,
+                 MessagingConstants.PushNotificationDetails.IDENTITY: [
+                    MessagingConstants.PushNotificationDetails.NAMESPACE: [
+                        MessagingConstants.PushNotificationDetails.CODE: MessagingConstants.PushNotificationDetails.JsonValues.ECID
+                    ],
+                    MessagingConstants.PushNotificationDetails.ID: ecid
+                 ]]
+            ]
+        ]
+
+        // Creating xdm edge event data
+        let xdmEventData: [String: Any] = [MessagingConstants.XDM.DataKeys.DATA: profileEventData]
+        // Creating xdm edge event with request content source type
+        let event = Event(name: MessagingConstants.EventName.PUSH_PROFILE_EDGE,
+                          type: EventType.edge,
+                          source: EventSource.requestContent,
+                          data: xdmEventData)
+        dispatch(event: event)
+    }
+
+    /// Get platform based on the `messaging.useSandbox` config value
+    /// - Parameters:
+    ///     - config: `[AnyHashable: Any]` with platform informations
+    func getPlatform(config: [AnyHashable: Any]) -> String {
+        return config[MessagingConstants.SharedState.Configuration.USE_SANDBOX] as? Bool ?? false
+            ? MessagingConstants.PushNotificationDetails.JsonValues.APNS_SANDBOX
+            : MessagingConstants.PushNotificationDetails.JsonValues.APNS
+    }
+
+    // MARK: - private methods
+
+    /// Adding Adobe/AJO specific data to tracking information map.
+    ///
+    /// - Parameters:
+    ///  - event: `Event` with Adobe AJO tracking information
+    ///  - xdmDict: `[String: Any]` which is updated with the AJO tracking information.
+    /// - Returns: a dictionary combining Adobe related data with the provided `xdmDict`
+    private func addAdobeData(event: Event, xdmDict: [String: Any]) -> [String: Any] {
+        // make sure this event has adobe xdm data
+        guard event.adobeXdm != nil else {
+            Log.warning(label: MessagingConstants.LOG_TAG,
+                        "Failed to update xdmMap with Adobe/AJO related informations : Adobe/AJO information are invalid or not available in the event '\(event.id.uuidString)'.")
+            return xdmDict
+        }
+
+        // required keys are found using the following priority
+        // 1. check the event's "mixins" key
+        // 2. check the event's "cjm" key
+        var mixins: [String: Any]
```

Comment:
we can leverage the power of `guard` god here
```
let mixins = event.mixins, !mixins.empty else {
  // log
  return xdmDict
}
```

Reply:
we don't actually want to return if mixins is nil or empty - we want to check to see if `cjm` has any content (see the comment in code for clarification :) )